### PR TITLE
Add a function to remove a registered gesture recognizer in gestures

### PR DESCRIPTION
**Pull request by cathyxz**
Closes https://github.com/ampproject/amphtml/issues/12112. 

This is a small PR. Please review the API for this new function and give feedback on whether it seems reasonable. The reason I'm proposing this function is because using `SwipeXY` to pan on `image-viewer` conflicts with using `SwipeY` to swipe-to-close on `amp-lightbox-viewer`. We want to deregister the `SwipeXY` on `image-viewer` when the image isn't scaled. 

This PR enables swipe up to close lightbox, prototype on [feature branch](https://github.com/cathyxz/amphtml/tree/feature/lightbox-swipe-to-close), deployment in progress on [heroku](https://amp-cathyxz.herokuapp.com/test/manual/amp-lightbox-viewer-optin.amp.html) for iOS testing. The actual lightbox swipe to close feature will be in a separate PR that is based off of https://github.com/ampproject/amphtml/pull/12256.

### CODE REVIEW

**Comment by cathyxz** [comment]
Merging master onto this branch for test fix in https://github.com/ampproject/amphtml/pull/12272.

File: src/gesture.js [hunk 1]
```diff
@@ -173,6 +175,35 @@ export class Gestures {
...
+   * Unsubscribes all handlers from the given gesture recognizer. Returns
+   * true if anything was done. Returns false if there were no handlers
+   * registered on the given gesture recognizer in first place.
+   *
+   * @param {function(new:GestureRecognizer<DATA>, !Gestures)} recognizerConstr
+   * @return {boolean}
+   */
+  removeGesture(recognizerConstr) {
+    const type = new recognizerConstr(this).getType();
+    const overserver = this.overservers_[type];
```
Comment (cvializ) [inline src/gesture.js]: It'd be nice if we could get the type of the constructor without creating an instance, but it doesn't look like there's another way to do it right now without refactoring, maybe adding an inherited static method on the subclasses to replace the hardcoded string in the constructor. Not necessary to fix, just thinking out loud.

File: src/gesture.js [hunk 2]
```diff
@@ -173,6 +175,35 @@ export class Gestures {
...
 
+  /**
+   * Unsubscribes all handlers from the given gesture recognizer. Returns
+   * true if anything was done. Returns false if there were no handlers
+   * registered on the given gesture recognizer in first place.
+   *
+   * @param {function(new:GestureRecognizer<DATA>, !Gestures)} recognizerConstr
+   * @return {boolean}
+   */
+  removeGesture(recognizerConstr) {
```
Comment (aghassemi) [inline src/gesture.js]: Could this become a  `remove` (or `unlisten`) on `GestureRecognizer` class and hence on every recognizer instance?

My concern with current approach is that it gives one part of the code to unregister handlers for everyone which breaks code isolation.

Is there a use-case that unregistering handlers individually is not enough?

Comment (aghassemi) [reply src/gesture.js]: chatted offline, `gesture` instance is tied to the `element` so this is fine.

File: src/gesture.js (hunk 1 above)
Comment (cathyxz) [reply src/gesture.js]: Can investigate feasibility and add a GFI task to refactor.

File: src/gesture.js (hunk 2 above)
Comment (cathyxz) [reply src/gesture.js]: Anymore changes to address or are we good for merge? =)

Comment (aghassemi) [reply src/gesture.js]: good to go, merged,

**Pull request merged**